pkg/auxnetdevice/auxNetDeviceProvider.go

```diff
@@ -0,0 +1,193 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2022 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package auxnetdevice
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/golang/glog"
+	"github.com/jaypipes/ghw"
+
+	"github.com/k8snetworkplumbingwg/sriov-network-device-plugin/pkg/types"
+	"github.com/k8snetworkplumbingwg/sriov-network-device-plugin/pkg/utils"
+)
+
+const (
+	maxVendorNameLen  = 20
+	maxProductNameLen = 40
+	classIDBaseInt    = 16
+	classIDBitSize    = 64
+)
+
+type auxNetDeviceProvider struct {
+	deviceList []*ghw.PCIDevice
+	rFactory   types.ResourceFactory
+}
+
+// NewAuxNetDeviceProvider DeviceProvider implementation from auxNetDeviceProvider instance
+func NewAuxNetDeviceProvider(rf types.ResourceFactory) types.DeviceProvider {
+	return &auxNetDeviceProvider{
+		rFactory:   rf,
+		deviceList: make([]*ghw.PCIDevice, 0),
+	}
+}
+
+func (ap *auxNetDeviceProvider) GetDiscoveredDevices() []*ghw.PCIDevice {
+	return ap.deviceList
+}
+
+func (ap *auxNetDeviceProvider) GetDevices(rc *types.ResourceConfig) []types.HostDevice {
+	newAuxDevices := make([]types.HostDevice, 0)
+	for _, device := range ap.deviceList {
+		// discover auxiliary device names
+		auxDevs, err := utils.GetAuxNetDevicesFromPci(device.Address)
+		if err == nil {
+			for _, auxDev := range auxDevs {
```

Comment:
can auxDevs  be empty? if yes we should log a warning that there is no auxDevs  for device.Address

Reply:
Yes